Core/GDCore/Project/Project.cpp

```diff
@@ -671,16 +703,15 @@ void Project::UnserializeFrom(const SerializerElement& element) {
   if (currentPlatform == NULL && !platforms.empty())
     currentPlatform = platforms.back();
 
-  // TODO EBO All events based behaviors and events based objects should be
-  // added to the project before the content of events based objects is
-  // unserialized because when unserializing child-objects InsertNewObject and
-  // AddNewBehavior need to get EventsBasedObject and EventsBasedBehavior
-  // instances from the Project.
   eventsFunctionsExtensions.clear();
   const SerializerElement& eventsFunctionsExtensionsElement =
       element.GetChild("eventsFunctionsExtensions");
   eventsFunctionsExtensionsElement.ConsiderAsArrayOf(
       "eventsFunctionsExtension");
+  // First, only unserialize behaviors and objects names.
+  // As event based objects can contains CustomObject and Custom Object,
+  // this allows them to reference EventBasedBehavior and EventBasedObject
+  // respectively.

```

Comment:
With these 2 passes, we won't have any issues to unserialize extensions that depends on each other.
It could be done only for EventBasedObject but I find it clearer to do it for behavior and free functions too.